include/ideep/operators/conv.hpp

```diff
@@ -20,26 +20,36 @@ struct convolution_forward_params {
   convolution_forward_params(
       dnnl::convolution_forward::primitive_desc&& pd,
       dnnl::convolution_forward&& primitive,
+      attr_t&& op_attr,
       int groups)
       : pd(std::move(pd)),
         primitive(std::move(primitive)),
+        op_attr(op_attr),
         groups(groups),
-        bias_attr(attr_t()) {}
+        bias_attr(attr_t()),
+        pd_use_threads(omp_get_max_threads()) {}
 
   convolution_forward_params(
       dnnl::convolution_forward::primitive_desc&& pd,
       dnnl::convolution_forward&& primitive,
+      attr_t&& op_attr,
       int groups,
       attr_t&& bias_attr)
       : pd(std::move(pd)),
         primitive(std::move(primitive)),
+        op_attr(op_attr),
         groups(groups),
-        bias_attr(std::move(bias_attr)) {}
+        bias_attr(std::move(bias_attr)),
+        pd_use_threads(omp_get_max_threads()) {}
 
   dnnl::convolution_forward::primitive_desc pd;
   dnnl::convolution_forward primitive;
+  attr_t op_attr;
   int groups;
   attr_t bias_attr;
+  // From IPEX. Set in `do_prepare()`, only used outside ideep.
+  // TO-DO: Use a better name, i.e., num_threads
+  int pd_use_threads;
```

Comment:
We only have it for conv but not matmul?

Reply:
No, only for conv. In IPEX, only conv has the struct of forward parameters. For other ops, computation is done directly.